exe/wallet/Main.hs

```diff
@@ -201,20 +227,40 @@ exec manager args
         )
         = client (Proxy @("v2" :> Api))
 
-    runClient :: ToJSON a => ClientM a -> IO ()
-    runClient cmd = do
+    -- | 'runClient' requires a type-application to carry a particular
+    -- namespace and adjust error messages accordingly. For instances, when
+    -- running commands from the 'cardano-wallet wallet' namespace, one should
+    -- do:
+    --
+    -- @
+    -- runClient @Wallet ...
+    -- @
+    runClient
+        :: forall b a. Typeable b
+        => (a -> BL.ByteString)
+        -> ClientM a
+        -> IO ()
+    runClient encode cmd = do
         port <- args `parseArg` longOption "port"
         let env = mkClientEnv manager (BaseUrl Http "localhost" port "")
         res <- runClientM cmd env
         case res of
+            Left (FailureResponse r) | responseStatusCode r == status404 -> do
+                let typ = T.pack $ tyConName $ typeRepTyCon $ typeRep $ Proxy @b
```

Comment:
:+1:

Reply:
I've added a similar one for `status409 :: Conflict` for already existing resources.